KesaApplication/ViewController.swift

```diff
@@ -10,16 +10,130 @@ import UIKit
+        emailLabel.textColor = UIColor.blackColor()
+        passwordLabel.textColor = UIColor.blackColor()
+    }
+    
+    
+    
+    // Highlighting each labels when their text fields are pressed
+    @IBAction func firstNameTextFieldPressed(sender: UITextField) {
+        firstNameLabel.textColor = UIColor.orangeColor()
+        lastNameLabel.textColor = UIColor.blackColor()
+        emailLabel.textColor = UIColor.blackColor()
+        passwordLabel.textColor = UIColor.blackColor()
+    }
+    
+    @IBAction func lastNameTextFieldPressed(sender: UITextField) {
+        firstNameLabel.textColor = UIColor.blackColor()
+        lastNameLabel.textColor = UIColor.orangeColor()
+        emailLabel.textColor = UIColor.blackColor()
+        passwordLabel.textColor = UIColor.blackColor()
     }
 
-
+    
+    @IBAction func emailTextFieldPressed(sender: UITextField) {
+        firstNameLabel.textColor = UIColor.blackColor()
+        lastNameLabel.textColor = UIColor.blackColor()
+        emailLabel.textColor = UIColor.orangeColor()
+        passwordLabel.textColor = UIColor.blackColor()
+    }
+    
+    @IBAction func passwordTextFieldPressed(sender: UITextField) {
+        firstNameLabel.textColor = UIColor.blackColor()
+        lastNameLabel.textColor = UIColor.blackColor()
+        emailLabel.textColor = UIColor.blackColor()
+        passwordLabel.textColor = UIColor.orangeColor()
+    }
+    
+    
+    
+    // When "Create Account" button is pressed
+    @IBAction func createAccountButtonPressed(sender: UIButton) {
+        
+        let userFirstName = firstName.text
+        let userLastName = lastName.text
+        let userEmail = email.text
+        let userPassword = password.text
+        
+        // Check for empty fields
+        if (userFirstName!.isEmpty || userLastName!.isEmpty ||
+            userEmail!.isEmpty || userPassword!.isEmpty) {
+            
+            // Display alert message 
+            displayAlertMessage ("All fields are required");
+            
+            return
+        }
+        
+        // Email validity check
+        if isValidEmail(userEmail!)  {
+        
+        
+            // Store data
+            
+        }
+        
+        else {
+            displayAlertMessage ("Invalid email address")
+            
+            return
+        }
+    }
+    
+    
+    
+    // Function for displaying alert messages
+    func displayAlertMessage (userMessage: String){
+        
+        let alert = UIAlertController(title: "Alert", message: userMessage, preferredStyle: UIAlertControllerStyle.Alert)
+        
+        let okAction = UIAlertAction(title: "Ok", style: UIAlertActionStyle.Default, handler: nil)
+        
+        alert.addAction(okAction)
+        
+        self.presentViewController(alert, animated: true, completion: nil)
+    }
+    
+    
+    
+    // Function to help check for the email validity 
+    // Returns a boolean
+    func isValidEmail (testEmail:String) -> Bool {
```

Comment:
Changed the function name to isValidEmail Format.

Made a new swift file for **Utility** class. The class has functions **isValidEmailFormat** and **isValidNameFormat**